src/ray/raylet/scheduling/scheduling_policy.h

```diff
@@ -70,6 +70,10 @@ class SchedulingPolicy {
   /// List of nodes in the clusters and their resources organized as a map.
   /// The key of the map is the node ID.
   const absl::flat_hash_map<int64_t, Node> &nodes_;
+  // The node to start round robin if it's spread scheduling.
+  // The index may be inaccurate when nodes are added or removed dynamically,
+  // but it should still be better than always scanning from 0 for spread scheduling.
```

Comment:
Hmm if we kept the node IDs in a sorted container (`std::set`, `absl::btree_set`, or `boost::flat_set`) that's properly updated upon node addition or removal, and we used that to keep `spread_scheduling_next_index_` updated, we could:
1. Keep `spread_scheduling_next_index_` accurate in the face of nodes being added and removed.
2. Obviate the need for a sort of all node IDs on the critical scheduling path. Since node membership changes will most likely be far less frequent than task scheduling, this could be a (small) scheduling performance win for large clusters.

Probably not worth the complexity for now.